src/AspNet.Security.OAuth.Introspection/Events/ValidateAudienceContext.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AspNet.Security.OAuth.Introspection {
+    public class ValidateAudienceContext : BaseIntrospectionContext {
+        public ValidateAudienceContext(
+            HttpContext context, 
+            OAuthIntrospectionOptions options,
+            JToken payload) 
+            : base(context, options) {
+            Payload = payload;
+            var audiences = payload[OAuthIntrospectionConstants.Claims.Audience];
```

Comment:
As for the constructor, I figured it made sense as part of the responsibility of the context to parse out the audiences since it was being passed the `payload` anyway, though I wonder if that should be `JToken` now or just pass in an audience string/audiences enumerable... Thoughts?